corehq/motech/repeaters/expression/repeaters.py

```diff
@@ -120,3 +170,23 @@ def available_for_domain(cls, domain):
             super(ArcGISFormExpressionRepeater, cls).available_for_domain(domain)
             and ARCGIS_INTEGRATION.enabled(domain)
         )
+
+
+def get_evaluation_context(domain, repeat_record, payload_doc, response):
+    try:
+        body = response.json()
+    except JSONDecodeError:
+        body = response.text
+    return EvaluationContext({
+        'domain': domain,
+        'success': is_success_response(response),
+        'payload': {
```

Comment:
nit: I believe these properties will now get used in the expression.
If I think of "payload" I assume it would be the doc itself. Keeping it as a different object with id and doc itself might get confusing.
So, the expression would become `payload.doc.name` instead of `payload.name` which seems a bit confusing.

Reply:
The tricky thing is that the actual payload sent by the repeater is determined by the UCR expression of the repeater so there's no way to know what might be in that payload.

Doing it this way guarantees a consistent and known structure. It also makes it easy to access the doc ID without having to know if it's a case or a form (which have different ID names).